include/mujincontrollerclient/mujincontrollerclient.h

```diff
@@ -139,6 +139,27 @@ typedef boost::shared_ptr<DebugResource> DebugResourcePtr;
 typedef boost::weak_ptr<DebugResource> DebugResourceWeakPtr;
 typedef double Real;
 
+/// \brief an attachment to a log entry
+struct LogEntryAttachment
+{
+    std::string filename; // filename
+    std::vector<unsigned char> data; // data for the attachment
+};
+
+typedef boost::shared_ptr<LogEntryAttachment> LogEntryAttachmentPtr;
+typedef boost::weak_ptr<LogEntryAttachment> LogEntryAttachmentWeakPtr;
+
+/// \brief a log entry in mujin controller
+struct LogEntry
+{
+    rapidjson::Document entry; // log entry data in JSON format
```

Comment:
Please do `rapidjson::Value rEntry` here, assuming controllerclientcpp does not need to modify things, you do not need a `Document` (that has allocator)

Reply:
We might need to inject "modifiedAt" into the logEntry

Reply:
@superfashi The API user should inject  `modifiedAt`, not the API.

Reply:
Oh, you are talking about the log entry uploader, I see. 

@ziyan Indeed we inject `modifiedAt` timestamp in the log entry uploader class and that class uses `LogEntry` as its input.

Reply:
I could make the uploader accept Document, and then construct log entry on the uploader side

Reply:
OK, we decided to accept documents in the uploader. This makes the C++ and Python implementations more similar, too.